src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs

```diff
@@ -405,6 +480,35 @@ internal static unsafe int WriteFileNative(SafeFileHandle handle, ReadOnlySpan<b
             }
         }
 
+        private static void GetPathForNtCreateFile(string fullPath, out ReadOnlySpan<char> prefixedAbsolutePath, out char[]? rentedArray)
+        {
+            const string mandatoryNtPrefix = @"\??\";
+
+            if (fullPath.StartsWith(mandatoryNtPrefix, StringComparison.Ordinal))
+            {
+                prefixedAbsolutePath = fullPath;
+                rentedArray = null;
+            }
+            else if (fullPath.StartsWith(@"\\?\", StringComparison.Ordinal)) // NtCreateFile does not support "\\?\" prefix, only "\??\"
```

Comment:
Should `@"\\?\"` be also a const?

Reply:
If the given string is used only once, the C# compiler is smart enough to treat it as const. Below you can see the IL for `\??\` (const) and `\\?\` (not const) being the same:

![obraz](https://user-images.githubusercontent.com/6011991/116378942-dc125a00-a812-11eb-86cf-6388aaffa017.png)